src/agent/src/main.rs

```diff
@@ -96,6 +96,13 @@ fn main() -> Result<()> {
     lazy_static::initialize(&AGENT_CONFIG);
     let agentConfig = AGENT_CONFIG.clone();
     let mut config = agentConfig.write().unwrap();
+
+    if unistd::getpid() == Pid::from_raw(1) {
+        let writer = io::stdout();
+        let logger = logging::create_logger(NAME, "agent", slog::Level::Debug, writer);
```

Comment:
Hi @lifupan - can you add more details to the commit explaining why this is necessary With this code, the logger will be recreated when the agent is running as PID 1 which isn't ideal.

Reply:
Hi @jodh-intel 

Since the config.parse_cmdline will read /proc/cmdline to get the agent options such agent.log etc, but before do the init_agent_as_init(), in which it will do the base  mount such as mount /proc etc, thus  it cannot access /proc/cmdline.  Thus we should do the init_agent_as_init() before call  config.parse_cmdline(). But init_agent_as_init() also needs a logger, thus there will be and old puzzle of Chicken and egg. That's why we need to create the logger twice.

Reply:
Thanks @lifupan - I see the problem now :smile:

It might be worth adding a comment on this first `logger` explaining why it's being created  here and then again later.